pyvrp/cpp/search/Route.cpp

```diff
@@ -28,7 +28,11 @@ Route::NodeStats::NodeStats(TimeWindowSegment const &tws)
 {
 }
 
-std::pair<double, double> const &Route::centroid() const { return centroid_; }
+std::pair<double, double> const &Route::centroid() const
+{
+    assert(!dirty);
+    return centroid_;
+}
```

Comment:
Do we also need to add an assertion in `Route::overlapsWith`? I think it's possible there to compare "dirty" routes since the private member `centroid_` is used.

Reply:
Yes, good call. One sec.